synapseclient/models/protocols/access_control_protocol.py

```diff
@@ -0,0 +1,104 @@
+"""Protocol for the specific methods of this class that have synchronous counterparts
+generatred at runtime."""
+
+from typing import Dict, List, Optional, TYPE_CHECKING, Union, Protocol
+from synapseclient import Synapse
+
+if TYPE_CHECKING:
+    from synapseclient.core.models.permission import Permissions
+
+
+class AccessControllableSynchronousProtocol(Protocol):
+    """
+    The protocol for methods that are asynchronous but also
+    have a synchronous counterpart that may also be called.
+    """
+
+    def get_permissions(
+        self,
+        synapse_client: Optional[Synapse] = None,
+    ) -> "Permissions":
+        """
+        Get the [permissions][synapseclient.core.models.permission.Permissions]
+        that the caller has on an Entity.
+
+        Arguments:
+            synapse_client: If not passed in or None this will use the last client
+                from the `.login()` method.
+
+        Returns:
+            A Permissions object
+
+
+        Example: Using this function:
+            Getting permissions for a Synapse Entity
+
+                permissions = File(id="syn123").get_permissions()
+
+            Getting access types list from the Permissions object
+
+                permissions.access_types
+        """
+        return self
```

Comment:
I would have liked to keep a return off of this - However, whatever I tried would cause an error to pop up if it was not included.

![image](https://github.com/Sage-Bionetworks/synapsePythonClient/assets/17128019/aa5c9a9b-7e0d-400b-acc8-09e01b24e3cf)